src/docfx/build/Build.cs

```diff
@@ -68,27 +68,35 @@ bool ShouldBuildFile(Document file)
             }
         }
 
-        private static Task<DependencyMap> BuildFile(
+        private static async Task<DependencyMap> BuildFile(
             Context context,
             Document file,
             TableOfContentsMap tocMap,
             ContributionInfo contribution,
             Action<Document> buildChild)
         {
-            switch (file.ContentType)
+            try
             {
-                case ContentType.Asset:
-                    return BuildAsset(context, file);
-                case ContentType.Markdown:
-                    return BuildMarkdown.Build(context, file, tocMap, contribution, buildChild);
-                case ContentType.SchemaDocument:
-                    return BuildSchemaDocument.Build();
-                case ContentType.TableOfContents:
-                    return BuildTableOfContents.Build(context, file, tocMap, buildChild);
-                case ContentType.Redirection:
-                    return BuildRedirectionItem(context, file);
-                default:
-                    return Task.FromResult(DependencyMap.Empty);
+                switch (file.ContentType)
+                {
+                    case ContentType.Asset:
+                        return await BuildAsset(context, file);
+                    case ContentType.Markdown:
+                        return await BuildMarkdown.Build(context, file, tocMap, contribution, buildChild);
+                    case ContentType.SchemaDocument:
+                        return await BuildSchemaDocument.Build();
+                    case ContentType.TableOfContents:
+                        return await BuildTableOfContents.Build(context, file, tocMap, buildChild);
+                    case ContentType.Redirection:
+                        return await BuildRedirectionItem(context, file);
+                    default:
+                        return DependencyMap.Empty;
+                }
+            }
+            catch (DocfxException ex)
```

Comment:
>DocfxException [](start = 19, length = 14)

which ex should be thrown and which ex whould be handled in their own pipeline? should we handle all errors in their own pipeline?